libfwupd/fwupd-client-sync.c

```diff
@@ -2709,3 +2709,60 @@ fwupd_client_emulation_save(FwupdClient *self, GCancellable *cancellable, GError
 	}
 	return g_steal_pointer(&helper->bytes);
 }
+
+static void
+fwupd_client_security_harden_cb(GObject *source, GAsyncResult *res, gpointer user_data)
+{
+	FwupdClientHelper *helper = (FwupdClientHelper *)user_data;
+	helper->ret =
+	    fwupd_client_security_harden_finish(FWUPD_CLIENT(source), res, &helper->error);
+	g_main_loop_quit(helper->loop);
+}
+
+/**
+ * fwupd_client_security_harden:
+ * @self: a #FwupdClient
+ * @appstream_id: the AppStream_id
+ * @enable: TRUE to enable hardening, FALSE to disable
+ * @cancellable: (nullable): optional #GCancellable
+ * @error: (nullable): optional return location for an error
+ *
+ * Harden the security attributes.
+ *
+ * Returns: %TRUE for success
+ *
+ * Since: 1.9.4
+ **/
+gboolean
+fwupd_client_security_harden(FwupdClient *self,
+			     const gchar *appstream_id,
+			     gboolean enable,
+			     GCancellable *cancellable,
+			     GError **error)
+{
+	g_autoptr(FwupdClientHelper) helper = NULL;
+
+	g_return_val_if_fail(FWUPD_IS_CLIENT(self), FALSE);
+	g_return_val_if_fail(appstream_id != NULL, FALSE);
+	g_return_val_if_fail(cancellable == NULL || G_IS_CANCELLABLE(cancellable), FALSE);
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+	/* connect */
+	if (!fwupd_client_connect(self, cancellable, error))
+		return FALSE;
```

Comment:
I think this should be responsible by the caller, not by `fwupd_client_security_harden`

Reply:
Do you mean this?
https://github.com/fwupd/fwupd/blob/54ad2cca592f6de83609797b34292808e348385d/src/fu-util.c#L5187

Move fwupd_client_connect() to the caller.